modules/bastion/assets/cloudinit_centos_8.cfg.tpl

```diff
@@ -0,0 +1,59 @@
+## template: jinja
+#cloud-config
+package_update: true
+package_upgrade: true
+package_reboot_if_required: true
+write_files:
+- path: /etc/sysctl.d/10-ip-forwarding.conf
+  content: |
+    net.ipv4.ip_forward=1
+- path: /etc/iptables/rules.v4
+  content: |
+    # Generated by iptables-save v1.8.4 on Tue Aug 24 19:16:37 2021
+    *filter
+    :INPUT ACCEPT [0:0]
+    :FORWARD ACCEPT [0:0]
+    :OUTPUT ACCEPT [0:0]
+    COMMIT
+    # Completed on Tue Aug 24 19:16:37 2021
+    # Generated by iptables-save v1.8.4 on Tue Aug 24 19:16:37 2021
+    *nat
+    :PREROUTING ACCEPT [0:0]
+    :INPUT ACCEPT [0:0]
+    :OUTPUT ACCEPT [0:0]
+    :POSTROUTING ACCEPT [0:0]
+    -A POSTROUTING -o bond0 -j MASQUERADE
+    COMMIT
+    # Completed on Tue Aug 24 19:16:37 2021
+- path: /etc/iptables/rules.v6
+  content: |
+    # Generated by ip6tables-save v1.8.4 on Tue Aug 24 19:16:37 2021
+    *filter
+    :INPUT ACCEPT [0:0]
+    :FORWARD ACCEPT [0:0]
+    :OUTPUT ACCEPT [0:0]
+    COMMIT
+- path: /etc/network/interfaces
```

Comment:
As a starting point for changes to consider, ChatGPT suggests:

```yaml
## template: jinja
#cloud-config
package_update: true
package_upgrade: true
package_reboot_if_required: true
write_files:
- path: /etc/sysctl.d/10-ip-forwarding.conf
  content: |
    net.ipv4.ip_forward=1
- path: /etc/sysconfig/iptables
  content: |
    # Generated by iptables-save v1.8.4 on Tue Aug 24 19:16:37 2021
    *filter
    :INPUT ACCEPT [0:0]
    :FORWARD ACCEPT [0:0]
    :OUTPUT ACCEPT [0:0]
    COMMIT
    # Completed on Tue Aug 24 19:16:37 2021
    # Generated by iptables-save v1.8.4 on Tue Aug 24 19:16:37 2021
    *nat
    :PREROUTING ACCEPT [0:0]
    :INPUT ACCEPT [0:0]
    :OUTPUT ACCEPT [0:0]
    :POSTROUTING ACCEPT [0:0]
    -A POSTROUTING -o bond0 -j MASQUERADE
    COMMIT
    # Completed on Tue Aug 24 19:16:37 2021
- path: /etc/sysconfig/ip6tables
  content: |
    # Generated by ip6tables-save v1.8.4 on Tue Aug 24 19:16:37 2021
    *filter
    :INPUT ACCEPT [0:0]
    :FORWARD ACCEPT [0:0]
    :OUTPUT ACCEPT [0:0]
    COMMIT
- path: /etc/sysconfig/network-scripts/ifcfg-bond0.${metal_vlan_id}
  content: |
    DEVICE=bond0.${metal_vlan_id}
    BOOTPROTO=static
    ONBOOT=yes
    IPADDR=${address}
    NETMASK=${netmask}
    VLAN=yes
- path: /etc/dnsmasq.d/openshift.config
  content: |
    bind-interfaces
    interface=bond0.${metal_vlan_id}
    dhcp-range=${host_dhcp_start},${host_dhcp_end},${lease_time}
packages:
  - iptables
  - iptables-services
  - dnsmasq
runcmd:
- sysctl -p /etc/sysctl.d/10-ip-forwarding.conf
- systemctl restart network
- systemctl restart dnsmasq
```

Reply:
I don't know if we'll have eth0/eth1 names. On the bootstrap node this may be tricky unless we can disable predictable names for the first boot (Arch guidance: https://wiki.archlinux.org/title/Network_configuration#Revert_to_traditional_interface_names)

We'll also need to add kernel command line args for the IPXE scripts to have them come up bonded, which may also need to include `net.ifnames=0`.

Reply:
Alternatively, `enp0sX` naming may work fine if we document the plan configurations that these names are known to work within.